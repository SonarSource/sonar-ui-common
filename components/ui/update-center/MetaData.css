/*
 * Sonar UI Common
 * Copyright (C) 2019-2019 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */

.update-center-meta-data {
  margin: 16px 0;
  padding: 16px 16px 8px 16px;
  background: #f9f9fb;
  border: 1px solid #e6e6e6;
  border-radius: 3px;
}

.update-center-meta-data a svg {
  margin-right: 8px;
}

.update-center-meta-data-header {
  border-bottom: 1px solid #cfd3d7;
  padding-bottom: 16px;
}

.update-center-meta-data-header,
.update-center-meta-data-version-release-info,
.update-center-meta-data-version-links {
  display: flex;
}

.update-center-meta-data-header > * + *,
.update-center-meta-data-version-release-info > * + * {
  margin-left: 16px;
}

.update-center-meta-data-header > * + * {
  padding-left: 16px;
  border-left: 1px solid #cfd3d7;
}

.update-center-meta-data-versions {
  margin-top: 16px;
}

.update-center-meta-data-versions-show-more {
  font-size: 14px;
  float: right;
  color: #51575a;
  border-color: #7b8184;
  border-width: 0 0 1px 0;
  padding-left: 0;
  padding-right: 0;
  background: transparent;
  cursor: pointer;
}

.update-center-meta-data-versions-show-more:hover {
  color: #2d3032;
  border-color: #2d3032;
}

.update-center-meta-data-version {
  margin-bottom: 16px;
}

.update-center-meta-data-version + .update-center-meta-data-version {
  padding-top: 8px;
  border-top: 1px dashed #cfd3d7;
}

.update-center-meta-data-version-version {
  font-weight: bold;
  font-size: 18px;
}

.update-center-meta-data-version-release-info {
  margin-top: 8px;
  font-style: italic;
}

.update-center-meta-data-version-release-description {
  margin-top: 8px;
}

.update-center-meta-data-version-download > a,
.update-center-meta-data-version-release-notes > a {
  display: inline-block;
  margin: 8px 16px 0 0;
}
