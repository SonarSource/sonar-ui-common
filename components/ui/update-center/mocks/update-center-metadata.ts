/*
 * Sonar UI Common
 * Copyright (C) 2019-2020 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */

import { MetaDataInformation, MetaDataVersionInformation } from '../update-center-metadata';

export function mockMetaDataVersionInformation(
  overrides?: Partial<MetaDataVersionInformation>
): MetaDataVersionInformation {
  return {
    version: '5.13',
    date: '2019-05-31',
    compatibility: '6.7',
    archived: false,
    downloadURL: 'https://example.com/sonar-java-plugin-5.13.0.18197.jar',
    changeLogUrl: 'https://example.com/sonar-java-plugin/release',
    ...overrides
  };
}

export function mockMetaDataInformation(
  overrides?: Partial<MetaDataInformation>
): MetaDataInformation {
  return {
    name: 'SonarJava',
    key: 'java',
    isSonarSourceCommercial: true,
    organization: {
      name: 'SonarSource',
      url: 'http://www.sonarsource.com/'
    },
    category: 'Languages',
    license: 'SonarSource',
    issueTrackerURL: 'https://jira.sonarsource.com/browse/SONARJAVA',
    sourcesURL: 'https://github.com/SonarSource/sonar-java',
    versions: [
      mockMetaDataVersionInformation({ version: '2.0' }),
      mockMetaDataVersionInformation({ version: '1.0', archived: true })
    ],
    ...overrides
  };
}
