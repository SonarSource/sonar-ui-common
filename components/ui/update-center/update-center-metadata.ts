/*
 * Sonar UI Common
 * Copyright (C) 2019-2019 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */

export interface MetaDataInformation {
  category?: string;
  isSonarSourceCommercial?: boolean;
  issueTrackerURL?: string;
  key?: string;
  license?: string;
  name: string;
  organization?: {
    name: string;
    url?: string;
  };
  sourcesURL?: string;
  versions?: MetaDataVersionInformation[];
}

export interface MetaDataVersionInformation {
  archived?: boolean;
  changeLogUrl?: string;
  compatibility?: string;
  date?: string;
  description?: string;
  downloadURL?: string | AdvancedDownloadUrl[];
  version: string;
}

export interface AdvancedDownloadUrl {
  label?: string;
  url: string;
}
