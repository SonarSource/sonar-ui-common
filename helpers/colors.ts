/*
 * Sonar UI Common
 * Copyright (C) 2019-2020 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
/* eslint-disable no-bitwise, no-mixed-operators */
export function stringToColor(str: string) {
  let hash = 0;
  for (let i = 0; i < str.length; i++) {
    hash = str.charCodeAt(i) + ((hash << 5) - hash);
  }
  let color = '#';
  for (let i = 0; i < 3; i++) {
    const value = (hash >> (i * 8)) & 0xff;
    color += ('00' + value.toString(16)).substr(-2);
  }
  return color;
}

export function isDarkColor(color: string) {
  color = color.substr(1);
  if (color.length === 3) {
    // shortcut notation: #f90
    color = color[0] + color[0] + color[1] + color[1] + color[2] + color[2];
  }
  const rgb = parseInt(color.substr(1), 16);
  const r = (rgb >> 16) & 0xff;
  const g = (rgb >> 8) & 0xff;
  const b = (rgb >> 0) & 0xff;
  const luma = 0.2126 * r + 0.7152 * g + 0.0722 * b;
  return luma < 140;
}

export function getTextColor(background: string, dark = '#222', light = '#fff') {
  return isDarkColor(background) ? light : dark;
}
