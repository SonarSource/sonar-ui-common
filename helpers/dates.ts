/*
 * Sonar UI Common
 * Copyright (C) 2019-2019 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import * as parse from 'date-fns/parse';

function pad(number: number) {
  if (number < 10) {
    return '0' + number.toString();
  }
  return number;
}

type ParsableDate = string | number | Date;

export function parseDate(rawDate: ParsableDate): Date {
  return parse(rawDate);
}

export function toShortNotSoISOString(rawDate: ParsableDate): string {
  const date = parseDate(rawDate);
  return `${date.getFullYear()}-${pad(date.getMonth() + 1)}-${pad(date.getDate())}`;
}

export function toNotSoISOString(rawDate: ParsableDate): string {
  const date = parseDate(rawDate);
  return date.toISOString().replace(/\..+Z$/, '+0000');
}

export function isValidDate(date: Date): boolean {
  return !isNaN(date.getTime());
}
