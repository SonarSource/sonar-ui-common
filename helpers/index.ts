/*
 * Sonar UI Common
 * Copyright (C) 2019-2020 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */

export * from './colors';
export * from './cookies';
export * from './csv';
export * from './dates';
export * from './getHistory';
export * from './handleRequiredAuthentication';
export * from './init';
export * from './keycodes';
export * from './l10n';
export * from './measures';
export * from './pages';
export * from './path';
export * from './query';
export * from './ratings';
export * from './request';
export * from './scrolling';
export * from './search';
export {
  get as getFromLocalStorage,
  remove as removeFromLocalStorage,
  save as saveInLocalStorage,
} from './storage';
export * from './strings';
export * from './testUtils';
export * from './urls';
