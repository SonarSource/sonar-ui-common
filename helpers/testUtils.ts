/*
 * Sonar UI Common
 * Copyright (C) 2019-2019 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */

import { ReactWrapper, ShallowWrapper } from 'enzyme';

export function mockEvent(overrides = {}) {
  return {
    target: { blur() {} },
    currentTarget: { blur() {} },
    preventDefault() {},
    stopPropagation() {},
    ...overrides
  } as any;
}

export function click(element: ShallowWrapper | ReactWrapper, event = {}): void {
  // `type()` returns a component constructor for a composite element and string for DOM nodes
  if (typeof element.type() === 'function') {
    element.prop<Function>('onClick')();
    // TODO find out if `root` is a public api
    // https://github.com/airbnb/enzyme/blob/master/packages/enzyme/src/ReactWrapper.js#L109
    (element as any).root().update();
  } else {
    element.simulate('click', mockEvent(event));
  }
}

export function clickOutside(event = {}): void {
  const dispatchedEvent = new MouseEvent('click', event);
  window.dispatchEvent(dispatchedEvent);
}

export function submit(element: ShallowWrapper | ReactWrapper): void {
  element.simulate('submit', {
    preventDefault() {}
  });
}

export function change(element: ShallowWrapper | ReactWrapper, value: string, event = {}): void {
  // `type()` returns a component constructor for a composite element and string for DOM nodes
  if (typeof element.type() === 'function') {
    element.prop<Function>('onChange')(value);
    // TODO find out if `root` is a public api
    // https://github.com/airbnb/enzyme/blob/master/packages/enzyme/src/ReactWrapper.js#L109
    (element as any).root().update();
  } else {
    element.simulate('change', {
      target: { value },
      currentTarget: { value },
      ...event
    });
  }
}

export const KEYCODE_MAP: { [keycode: number]: string } = {
  13: 'enter',
  37: 'left',
  38: 'up',
  39: 'right',
  40: 'down'
};

export function keydown(key: number | string): void {
  let keyCode;
  if (typeof key === 'number') {
    keyCode = key;
  } else {
    // eslint-disable-next-line no-console
    console.warn('Using strings in keydown() is deprecated. Consider using the KeyCodes enum.');
    const mapped = Object.entries(KEYCODE_MAP).find(([_, value]) => value === key);
    if (!mapped) {
      throw new Error(`Cannot map key "${key}" to a keyCode!`);
    }
    keyCode = mapped[0];
  }

  const event = new KeyboardEvent('keydown', { keyCode, which: keyCode } as KeyboardEventInit);
  document.dispatchEvent(event);
}

export function elementKeydown(element: ShallowWrapper, keyCode: number): void {
  const event = {
    currentTarget: { element },
    keyCode,
    preventDefault() {}
  };

  if (typeof element.type() === 'string') {
    // `type()` is string for native dom elements
    element.simulate('keydown', event);
  } else {
    element.prop<Function>('onKeyDown')(event);
  }
}

export function resizeWindowTo(width?: number, height?: number) {
  // `document.documentElement.clientHeight/clientWidth` are getters by default,
  // so we need to redefine them. Pass `configurable: true` to allow to redefine
  // the properties multiple times.
  if (width) {
    Object.defineProperty(document.documentElement, 'clientWidth', {
      configurable: true,
      value: width
    });
  }
  if (height) {
    Object.defineProperty(document.documentElement, 'clientHeight', {
      configurable: true,
      value: height
    });
  }

  const resizeEvent = new Event('resize');
  window.dispatchEvent(resizeEvent);
}

export function scrollTo({ left = 0, top = 0 }) {
  Object.defineProperty(window, 'pageYOffset', { value: top });
  Object.defineProperty(window, 'pageXOffset', { value: left });
  const resizeEvent = new Event('scroll');
  window.dispatchEvent(resizeEvent);
}

export function setNodeRect({ width = 50, height = 50, left = 0, top = 0 }) {
  const { findDOMNode } = require('react-dom');
  const element = document.createElement('div');
  Object.defineProperty(element, 'getBoundingClientRect', {
    value: () => ({ width, height, left, top })
  });
  findDOMNode.mockReturnValue(element);
}

export function doAsync(fn?: Function): Promise<void> {
  return new Promise(resolve => {
    setImmediate(() => {
      if (fn) {
        fn();
      }
      resolve();
    });
  });
}

export async function waitAndUpdate(wrapper: ShallowWrapper<any, any> | ReactWrapper<any, any>) {
  await new Promise(setImmediate);
  wrapper.update();
}
