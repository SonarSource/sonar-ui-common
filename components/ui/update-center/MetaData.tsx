/*
 * Sonar UI Common
 * Copyright (C) 2019-2019 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */

import * as React from 'react';
import { getJSON } from '../../../helpers/request';
import './MetaData.css';
import MetaDataVersions from './MetaDataVersions';
import { MetaDataInformation } from './update-center-metadata';

interface Props {
  updateCenterKey?: string;
}

interface State {
  data?: MetaDataInformation;
}

export default class MetaData extends React.Component<Props, State> {
  mounted = false;
  state: State = {};

  componentDidMount() {
    this.mounted = true;
    this.fetchData();
  }

  componentDidUpdate(prevProps: Props) {
    if (prevProps.updateCenterKey !== this.props.updateCenterKey) {
      this.fetchData();
    }
  }

  componentWillUnmount() {
    this.mounted = false;
  }

  fetchData() {
    const { updateCenterKey } = this.props;

    if (updateCenterKey) {
      getJSON(`https://update.sonarsource.org/${updateCenterKey}.json`).then(
        (data: MetaDataInformation) => {
          if (this.mounted) {
            this.setState({ data });
          }
        },
        () => this.setState({ data: undefined })
      );
    } else {
      this.setState({ data: undefined });
    }
  }

  render() {
    const { data } = this.state;

    if (!data) {
      return null;
    }

    const { isSonarSourceCommercial, issueTrackerURL, license, organization, versions } = data;

    let vendor;
    if (organization) {
      vendor = organization.name;
      if (organization.url) {
        vendor = (
          <a href={organization.url} rel="noopener noreferrer" target="_blank">
            {vendor}
          </a>
        );
      }
    }

    return (
      <div className="update-center-meta-data">
        <div className="update-center-meta-data-header">
          {vendor && <span className="update-center-meta-data-vendor">By {vendor}</span>}
          {license && <span className="update-center-meta-data-license">{license}</span>}
          {issueTrackerURL && (
            <span className="update-center-meta-data-issue-tracker">
              <a href={issueTrackerURL} rel="noopener noreferrer" target="_blank">
                Issue Tracker
              </a>
            </span>
          )}
          {isSonarSourceCommercial && (
            <span className="update-center-meta-data-supported">Supported by SonarSource</span>
          )}
        </div>
        {versions && versions.length > 0 && <MetaDataVersions versions={versions} />}
      </div>
    );
  }
}
