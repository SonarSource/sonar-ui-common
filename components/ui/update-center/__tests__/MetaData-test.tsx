/*
 * Sonar UI Common
 * Copyright (C) 2019-2020 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */

import { shallow } from 'enzyme';
import * as React from 'react';
import { waitAndUpdate } from '../../../../helpers/testUtils';
import MetaData from '../MetaData';
import { mockMetaDataInformation } from '../mocks/update-center-metadata';
import { MetaDataInformation } from '../update-center-metadata';

beforeAll(() => {
  window.fetch = jest.fn();
});

beforeEach(() => {
  jest.resetAllMocks();
});

it('should render correctly', async () => {
  const metaDataInfo = mockMetaDataInformation();
  mockFetchReturnValue(metaDataInfo);

  const wrapper = shallowRender();
  await waitAndUpdate(wrapper);
  expect(wrapper).toMatchSnapshot();
});

it('should render correctly with organization', async () => {
  const metaDataInfo = mockMetaDataInformation({
    organization: { name: 'test-org', url: 'test-org-url' },
  });
  mockFetchReturnValue(metaDataInfo);

  const wrapper = shallowRender();
  await waitAndUpdate(wrapper);
  expect(wrapper).toMatchSnapshot();
});

it('should not render anything if call for metadata fails', async () => {
  const metaDataInfo = mockMetaDataInformation();
  mockFetchReturnValue(metaDataInfo, 404);

  const wrapper = shallowRender();
  await waitAndUpdate(wrapper);
  expect(wrapper.type()).toBeNull();
});

it('should fetch metadata again if the update center key if modified', async () => {
  const metaDataInfo = mockMetaDataInformation();
  mockFetchReturnValue(metaDataInfo);

  const wrapper = shallowRender();
  await waitAndUpdate(wrapper);

  expect(window.fetch).toHaveBeenCalledTimes(1);

  mockFetchReturnValue(metaDataInfo);
  wrapper.setProps({ updateCenterKey: 'abap' });

  expect(window.fetch).toHaveBeenCalledTimes(2);
});

function shallowRender(props?: Partial<MetaData['props']>) {
  return shallow<MetaData>(<MetaData updateCenterKey="apex" {...props} />);
}

function mockFetchReturnValue(metaDataInfo: MetaDataInformation, status = 200) {
  (window.fetch as jest.Mock).mockResolvedValueOnce({ status, json: () => metaDataInfo });
}
