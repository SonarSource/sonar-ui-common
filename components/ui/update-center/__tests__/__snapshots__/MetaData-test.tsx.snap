// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div
  className="update-center-meta-data"
>
  <div
    className="update-center-meta-data-header"
  >
    <span
      className="update-center-meta-data-vendor"
    >
      By 
      <a
        href="http://www.sonarsource.com/"
        rel="noopener noreferrer"
        target="_blank"
      >
        SonarSource
      </a>
    </span>
    <span
      className="update-center-meta-data-license"
    >
      SonarSource
    </span>
    <span
      className="update-center-meta-data-issue-tracker"
    >
      <a
        href="https://jira.sonarsource.com/browse/SONARJAVA"
        rel="noopener noreferrer"
        target="_blank"
      >
        Issue Tracker
      </a>
    </span>
    <span
      className="update-center-meta-data-supported"
    >
      Supported by SonarSource
    </span>
  </div>
  <MetaDataVersions
    versions={
      Array [
        Object {
          "archived": false,
          "changeLogUrl": "https://example.com/sonar-java-plugin/release",
          "compatibility": "6.7",
          "date": "2019-05-31",
          "downloadURL": "https://example.com/sonar-java-plugin-5.13.0.18197.jar",
          "version": "2.0",
        },
        Object {
          "archived": true,
          "changeLogUrl": "https://example.com/sonar-java-plugin/release",
          "compatibility": "6.7",
          "date": "2019-05-31",
          "downloadURL": "https://example.com/sonar-java-plugin-5.13.0.18197.jar",
          "version": "1.0",
        },
      ]
    }
  />
</div>
`;

exports[`should render correctly with organization 1`] = `
<div
  className="update-center-meta-data"
>
  <div
    className="update-center-meta-data-header"
  >
    <span
      className="update-center-meta-data-vendor"
    >
      By 
      <a
        href="test-org-url"
        rel="noopener noreferrer"
        target="_blank"
      >
        test-org
      </a>
    </span>
    <span
      className="update-center-meta-data-license"
    >
      SonarSource
    </span>
    <span
      className="update-center-meta-data-issue-tracker"
    >
      <a
        href="https://jira.sonarsource.com/browse/SONARJAVA"
        rel="noopener noreferrer"
        target="_blank"
      >
        Issue Tracker
      </a>
    </span>
    <span
      className="update-center-meta-data-supported"
    >
      Supported by SonarSource
    </span>
  </div>
  <MetaDataVersions
    versions={
      Array [
        Object {
          "archived": false,
          "changeLogUrl": "https://example.com/sonar-java-plugin/release",
          "compatibility": "6.7",
          "date": "2019-05-31",
          "downloadURL": "https://example.com/sonar-java-plugin-5.13.0.18197.jar",
          "version": "2.0",
        },
        Object {
          "archived": true,
          "changeLogUrl": "https://example.com/sonar-java-plugin/release",
          "compatibility": "6.7",
          "date": "2019-05-31",
          "downloadURL": "https://example.com/sonar-java-plugin-5.13.0.18197.jar",
          "version": "1.0",
        },
      ]
    }
  />
</div>
`;
